test/lib/generate/file-paths-test.ts

```diff
@@ -267,6 +269,13 @@ describe('generate (file paths)', function () {
       expect(readFileSync('rules/list.md', 'utf8')).toMatchSnapshot();
       expect(readFileSync('rules/no-foo/no-foo.md', 'utf8')).toMatchSnapshot();
     });
+
+    it('generates the documentation using a function for pathRuleDoc', async function () {
+      await generate('.', {
+        pathRuleDoc: (ruleName) => join('rules', ruleName, `${ruleName}.md`),
+      });
+      expect(readFileSync('rules/no-foo/no-foo.md', 'utf8')).toMatchSnapshot();
```

Comment:
I think we still want to snapshot test the README so we can make sure it links to the rule docs properly.

```suggestion
      expect(readFileSync('README.md', 'utf8')).toMatchSnapshot();
      expect(readFileSync('rules/no-foo/no-foo.md', 'utf8')).toMatchSnapshot();
```